cli/flox-activations/src/cli/activate/mod.rs

```diff
@@ -92,6 +114,112 @@ impl ActivateArgs {
         }
     }
 
+    fn startup_ctx(
+        ctx: ActivateCtx,
+        invocation_type: InvocationType,
+        env_diff: EnvDiff,
+        state_dir: &Path,
+    ) -> Result<StartupCtx> {
+        let is_sourcing_rc = std::env::var("_flox_sourcing_rc").is_ok_and(|val| val == "true");
+        let flox_activations = ctx.interpreter_path.join("libexec/flox-activations");
+        let self_destruct =
+            !std::env::var(STARTUP_SCRIPT_NO_SELF_DESTRUCT_VAR).is_ok_and(|val| val == "true");
```

Comment:
nit: I kinda think we should control this (and removing the ActivationContext file removal) with verbosity

Reply:
I don't see this as being related to verbosity at all. Why do you?

Reply:
I don't think it's intuitive that `-vvv` correlates to whether this file gets cleaned up, and I think it's valuable to be able to inspect the _exact_ file your shell sourced while you're inside the activation since running a script that you've printed isn't exactly the same thing as doing an activation (if you're debugging you want to be inspecting the same situation that the bug occurs in).